### DISCUSSION TITLE
v6: Accessing memory history from tests

### DISCUSSION THREAD

**Discussion by joshkel** [remix-run/react-router#8558] [Q&A] [answered]
With v5, I could create a memory history object, execute some code, and then inspect the results:

```js
const history = createMemoryHistory({ initialEntries });
render(
  <Router history={history}>
    <MyComponent />
  </Router>,
);
// ...do some stuff that triggers redirections, etc...
expect(history.location.search).toMatch(/desired/);
```

Is there any way to do something similar in React Router v6? I see where I can pass initial state in to MemoryRouter, but I don't see any way to get the current state back out.

**Comment by timdorr** [comment, marked as answer]
https://github.com/remix-run/react-router/issues/8264#issuecomment-991271554
